File: quad.java
import java.util.*;

public class Main{
	public static ArrayList<Double> quadratic(double a, double b, double c){
		ArrayList<Double> s = new ArrayList<>();
		if (b * b < 4 *a *c) return s;
		else if (b *b == 4 *a *c){
			double r = 2 * a;
			double t = 0 - b;
			double n = r / t;
			s.add(n);
			return s;
		}
		else{
			double r = 2 * a;
			double t = 0 - b;
			double n = b * b;
			double h = 4 *a * c;
			n -= h;
			n = Math.sqrt(n);
			double fi = (t + n) / r;
			double se = (t - n) / r;
			s.add(fi);
			s.add(se);
			return s;
		}
	}
	public static void main(String[] args){
		Scanner scan = new Scanner(System.in);
		double a = scan.nextDouble();
		double b = scan.nextDouble();
		double c = scan.nextDouble();
		if(a == 0){
			System.out.print("a cannot be equal to 0!);
			System.exit(0);
		}			 
		ArrayList<Double> x = quadratic(a, b, c);
		if (x.isEmpty()) System.out.print("IMAGINARY SOLUTIONS ARE NOT SUPPORTED!!!");
		else{
			for (int i = 0; i < x.size(); i++){
				System.out.print(x.get(i));
				if (i != x.size() - 1) System.out.print(" or ");
			}
		}
	}
}

File: quad.py
def quadratic(a,b,c):
    from math import sqrt
    s = []
    if b*b < 4 * a * c:
        return s
    elif b * b == 4 * a * c:
        r = 2 * a
        t = 0 - b
        n = r / t
        s.append(n)
        return s
    else:
        r = 2 * a
        t = 0 - b
        n = b * b
        h = 4 * a * c
        n -= h
        n = sqrt(n)
        fi = (t + n)/r
        se = (t - n)/r
        s.append(fi)
        s.append(se)
        return s
v = input().split()
if len(v) != 3:
    print("Invalid input")
    exit()
try:
    for i in range(3):
        v[i] = eval(v[i])
except:
    print("Invalid input")
    exit()
if v[0] == 0:
    print("a cannot be equal to 0!")
    exit()
n = quadratic(v[0],v[1],v[2])
if not n:
    print("IMAGINARY SOLUTIONS ARE NOT SUPPORTED!")
    exit()
for i in range(len(n)):
    print(n[i], end = "" )
    if i != len(n) - 1:
        print(" or ", end = "")
